src/core/js/models/lockingModel.js

```diff
@@ -0,0 +1,143 @@
+define([
+	'backbone'
+], function() {
+
+	var set = Backbone.Model.prototype.set;
+
+	_.extend(Backbone.Model.prototype, {
+
+		set: function(attrName, attrVal, options) {
+			var stopProcessing = !this.lockedAttributes || typeof attrName === "object" || typeof attrVal !== "boolean" || !this.isLocking(attrName);
+			if (stopProcessing) return set.apply(this, arguments);
+
+			var isSettingValueForSpecificPlugin = options && options.pluginName;
+			if (!isSettingValueForSpecificPlugin) {
+				console.error("Must supply a pluginName to change a locked attribute");
+				options.pluginName = "compatibility";
```

Comment:
It'll function fine, you'd just get the error as a warning. We could change it to a console.log if that makes it clearer.